README.md

```diff
@@ -47,23 +56,31 @@ Randomizes the capitalization of returned domain names.
 
 ## Dependencies
 
-This module depends on the `openssl` library and has an optional
-dependency on the `libidn` (with the [IDN](#idn) feature, enabled by
-default) library.
+Run-time dependencies:
+ * `openssl`
+ * `libidn` (if [IDN] is enabled)
+ * `libldns` (if [Internal ldns] is disabled)
+
+Compile-time dependencies (only when installing from source):
+ * `make`
+ * `Devel::CheckLib`
+ * `git` (if [Internal ldns] is enabled)
+ * `libtool` (if [Internal ldns] is enabled)
+ * `autoconf` (if [Internal ldns] is enabled)
+ * `automake` (if [Internal ldns] is enabled)
+
+Test-time dependencies:
+ * `Test::Fatal`
 
 
 ## Installation
```

Comment:
Move the "Installation" section up directly after the "Introduction" section. If you decide to follow the recommended way of installation you do not need to read the rest of this file.

Else it looks good.

Reply:
I agree with your point, but I also think the Installation, Post-installation sanity check and Testing sections should be within close proximity to each other. Do you think moving the Optional features section down to just above the Compatibility section would be an improvement over? I'm not so sure.

@sandoche2k Can you weigh in on this with some argument?

Reply:
@mattias-p, it would be an improvement to move "Optional features" down, but preferably "Recommeded installation" comes before "Dependicies" because if you follow the recommended installation you do always install `libidn` and internal LDNS.

Suggestion:

Break out "Recommended installation" and make that a section of highest level directly after "Introduction", and add a reference "For installation from source, read below". Then the rest could be as-is except that "Installation" is renamed to "Installation from source" (and no sub-level).

Reply:
I see no reason to streamline this document for people who just want to install Zonemaster. We're not directing those people here anyway. This installation documentation is primarily important for people who need non-default options - like packagers for instance.